File: JavaScriptBasics/WhileLoop/Exercise/Vacation.js
function Vacation(input) {
    let neededMoney = Number(input[0]);
    let startingBudget = Number(input[1]);
    let commandIndex = 2;
    let amountIndex = 3;
    let daysCounter = 0;
    let consecutiveSpending = 0;

    while (startingBudget < neededMoney && consecutiveSpending < 5) {
        let command = input[commandIndex];
        let amount = Number(input[amountIndex]);

        if (command === "spend") {
            startingBudget =startingBudget- amount;
            consecutiveSpending++;
        } else if (command === "save") {
            startingBudget =startingBudget+ amount;
            consecutiveSpending = 0;
        }

        daysCounter++;
        commandIndex = commandIndex+ 2;
        amountIndex = amountIndex+ 2;
    }

    if (consecutiveSpending === 5) {
        console.log("You can't save the money.");
        console.log(daysCounter);
    } else {
        console.log(`You saved the money for ${daysCounter} days.`);
    }
}

File: JavaBasics/WhileLoops/Exercise/Vacation.java
package EntryModule.WhileLoops.Exercise;

import java.util.Scanner;
public class Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double needed = Double.parseDouble(scanner.nextLine());
        double having = Double.parseDouble(scanner.nextLine());
        int inarow = 0;
        int days = 0;

        while(needed>having){
            days++;
            String move = scanner.nextLine();
            double sum = Double.parseDouble(scanner.nextLine());
            if(move.equals("spend")){sum=sum*(-1);}
            if(move.equals("spend")){inarow++;}else{inarow=0;}
            if(inarow==5){
                System.out.println("You can't save the money.");
                System.out.println(days);break;
            }
            having = having+sum;
            if(having<0){having=0;}
            if(having>=needed){
                System.out.println("You saved the money for "+days+" days.");
                break;}
        }


    }
}
